### PR TITLE
Config mgmt Get Nodes date filter

### PR DESCRIPTION
**Pull request by lancewf**
### :nut_and_bolt: Description: What code changed, and why?
Adding a start and end date to the GetNodes function to filter for when a node has last checked-in. 

### :chains: Related Resources

#3217 

### :athletic_shoe: How to Build and Test the Change
1. `build components/config-mgmt-service && build components/automate-gateway && start_all_services`
1. Add 3 nodes from now to 3 days ago 
```
for _ in {1..3} ; do
  date=$(date +%FT%TZ -d "$((RANDOM % 3)) day ago")
  generate_chef_run_failure_example | \
    jq --arg date "$date" '.start_time = $date | .end_time = $date' | \
    send_chef_data_raw
done
```
1. Add 2 nodes from over 6 days ago. 
```
  date=$(date +%FT%TZ -d "6 day ago")
  generate_chef_run_failure_example | \
    jq --arg date "$date" '.start_time = $date | .end_time = $date' | \
    send_chef_data_raw

  date=$(date +%FT%TZ -d "7 day ago")
  generate_chef_run_failure_example | \
    jq --arg date "$date" '.start_time = $date | .end_time = $date' | \
    send_chef_data_raw
```
1. Filter for nodes from 5 days to now and ensure only 3 nodes are returned 
```
date=$(date +%FT%TZ -d "5 day ago")
curl -s -f --insecure -H "api-token: $(get_api_token)" "https://localhost/api/v0/cfgmgmt/nodes?pagination.page=1&pagination.size=100&start=$date" | jq .[].id | wc -l
```
1. Filter for nodes up to 4 days ago and ensure only 2 nodes are returned. 
```
curl -s -f --insecure -H "api-token: $(get_api_token)" "https://localhost/api/v0/cfgmgmt/nodes?pagination.page=1&pagination.size=100&end=$date" | jq .[].id | wc -l
```

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: components/config-mgmt-service/params/parameters.go [hunk 1]
```diff
@@ -183,6 +183,43 @@ func ValidateDateRange(start string, end string) bool {
 	return true
 }
 
+// ValidateDateTimeRange will validate that the provided start & end date are valid. That means they
+// both have to have the right format and the start time must be less than or equal to the end time
+//
+// NOTE: If start or end are empty strings ("") that's consider an ok "empty" parameter
+func ValidateDateTimeRange(start, end string) bool {
```
Comment (lancewf) [inline components/config-mgmt-service/params/parameters.go:190]: Creates a new date validator for date time strings.

File: components/config-mgmt-service/backend/elastic/nodes.go [hunk 2]
```diff
@@ -191,23 +191,28 @@ func (es Backend) GetNodesPageByCursor(ctx context.Context, start time.Time,
...
 
 	// Adding the exists = true filter to the list of filters, because nodes
 	// have documents that persist in elasticsearch to hold historical data
 	// even after the node no longer exists
 	filters["exists"] = []string{"true"}
 
-	filtersQuery := newBoolQueryFromFilters(filters)
+	mainQuery := newBoolQueryFromFilters(filters)
+
+	rangeQuery, ok := newRangeQuery(startDate, endDate, NodeCheckin)
```
Comment (vjeffrey) [inline components/config-mgmt-service/backend/elastic/nodes.go:205]: does this mean filters are ignored when doing a range query?

Comment (lancewf) [reply components/config-mgmt-service/backend/elastic/nodes.go:205]: No, filters are added to the mainQuery and when there are date ranges it is added to the mainQuery also. 
I will add a test to ensure that is the case.

Comment (lancewf) [reply components/config-mgmt-service/backend/elastic/nodes.go:205]: Here is a test for that https://github.com/chef/automate/pull/3340/files#diff-d9bd977013f859bb8b84713df8afb223R413

**Pull request merged**